src/api/services/goals.service.js

```diff
@@ -72,29 +72,37 @@ exports.modifySubGoal = async (subGoal, subGoalId, mainGoalId) => {
   return await mainGoal.save();
 };
 
-exports.addUser = async (goalId, email) => {
-  const user = await User.findOne({ email }, "_id");
+exports.share = async (goalId, email) => {
+  const user = await User.findOne({ email });
   const mainGoal = await MainGoal.findById(goalId);
+  const isShareUser = mainGoal.users.includes(user._id);
+  const isShareGoal = user.createdGoals.includes(mainGoal._id);
```

Comment:
이미 공유 받은 유저인지만 확인해보면 되지 않을까요? 
저희가 DB를 믿을만하게 관리했다면 말이지만요. 👀

```suggestion
  const isSharedUser = mainGoal.users.includes(user._id);
```